docs/design/README.md

```diff
@@ -1827,6 +1827,13 @@ and starts with a letter. We intend to follow
 identifier characters, but a concrete set of valid characters has not been
 selected yet.
 
+The following names are reserved for internal Carbon implementation specifics,
+and to allow seamless interoperability with C++ codebases:
+
+-   Names with a double underscore anywhere are reserved;
+-   Names that begin with an underscore followed by an uppercase letter are
+    reserved.
+
```

Comment:
This feels a bit detailed for the overview. Can we make this a summary with fewer details, and link to the detailed design?

Reply:
Fixed - link to PR was already in the References section, so just wording adjusted.